### PR TITLE
Add GitHub Action

### PR DESCRIPTION
**Pull request by probonopd**
Add GitHub Action that builds and uploads wlmaker.

There may be some redundancies between which dependencies get installed via apt-get and git submodules; feel free to remove redundant deb packages.

### CODE REVIEW

**Comment by probonopd** [comment]
> This project is not an official Google project.

In this case, why is the @google-cla bot requesting me to sign a Contributor License Agreement (CLA)?

File: .github/workflows/build_and_upload_artifact.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        
+        - name: Checkout submodules
+          run: |
+               git config --global --add safe.directory /__w/wlmaker/wlmaker
+               git submodule update --init --recursive --merge
+
+        - name: Build dependencies
+          run: |
+               cd dependencies
+               export LD_LIBRARY_PATH="${HOME}/.local/lib/x86_64-linux-gnu"
```
Comment (phkaeser) [inline .github/workflows/build_and_upload_artifact.yml:36]: optional: portability issue -- you might change this to ${MACHTYPE}  (in bash).

(not sure it's worth it; but might be helpful to add a comment about it, so if somebody ever runs it on another architecture, they have a head-start in fixing)

**Review by phkaeser — APPROVED** [review]
Thanks for contributing the action!

**Comment by phkaeser** [comment]
Regarding the CLA: Since I'm employed by Google, I followed the default & recommended setup for license & github setup; that includes Apache2.0 as license, with Google LLC as copyright owner, and the CLA required to sign. 

I understand the company desires an agreement for contributions, since it comes into a codebase they have the copyright tacked on top. 

While I have a personal preference towards GPLv2, the Apache2.0 setup and CLA seemed fair enough for the purpose of wlmaker. I'm fine with the licence's permissiveness, and the CLA didn't strike me as prohibitive.

Do you see contentious issues with the cla/google requirement?

**Comment by probonopd** [comment]
> Do you see contentious issues with the cla/google requirement?

Haven't really checked it yet, it just struck me as odd since the project specifically mentions _not_ to be a Google project.

File: .github/workflows/build_and_upload_artifact.yml (hunk 1 above)
Comment (probonopd) [reply .github/workflows/build_and_upload_artifact.yml:36]: Thanks, on it.

Comment (probonopd) [reply .github/workflows/build_and_upload_artifact.yml:36]: `${MACHTYPE}` seems to resolve to `x86_64-pc-linux-gnu`, whereas we need `x86_64-linux-gnu`.

`dpkg-architecture -qDEB_HOST_MULTIARCH` seems to give us what we need, though.

Comment (phkaeser) [reply .github/workflows/build_and_upload_artifact.yml]: Oh, that's a subtle detail..! Well spotted & thanks for fixing here!

**Comment by phkaeser** [comment]
To make sure this isn't accidentally blocked on me: Change looks OK, and if I understand correctly, signing cla/google will greenlight it.

Is that OK for you to proceed, @probonopd ?

**Comment by probonopd** [comment]
Is there any chance to get this merged without creating a Google Account, which seems to be required for the CLA?

**Comment by phkaeser** [comment]
Looked through this in quite some detail, but: I'm afraid no. 

Due to international copyright & intellectual property situation, the company expects a CLA to be signed also for contributions to the personal projects of it's employees. And https://cla.developers.google.com/about is unfortunately clear in terms that it requires a Google account to sign the CLA.

While I'd want the bar for a contribution to be (virtually) non-existing, I tend to follow the rationale for a CLA (good write-up: https://julien.ponge.org/blog/in-defense-of-contributor-license-agreements/). 

Apologies that this turns out to be such a cumbersome process to contribute :-(

**Comment by probonopd** [comment]
Like https://github.com/phkaeser/wlmaker/pull/11, 

_I put the content of this PR into the public domain. CLA isn't an option for me due to the Google account requirement. Feel free to resubmit under different authorship (e.g., `git commit --amend --reset-author`) in order to satisfy CLA._

**Comment by phkaeser** [comment]
Thank you for the suggestion of moving to public domain. 

Will look into getting deployment merged; but may only get to it in 1-2w from now due to some personal time off. Apologies for the slow turnaround!

**Comment by phkaeser** [comment]
Heavily modelled from your input, https://github.com/phkaeser/wlmaker/pull/25 adds a build + test action on ubuntu-latest.

Doesn't upload an artefact, since dependencies also need to be included; #26 pending to (eventually) look into that.

With that, I'll close (and not merge) this pull request. Please re-open if you consider the artefact critical for your needs.

**Pull request closed without merging**